sway-core/src/parse_tree/expression/mod.rs

```diff
@@ -463,11 +479,23 @@ impl Expression {
                     ));
                 }
 
-                Expression::FunctionApplication {
+                let var_decls = arguments_buf
+                    .iter()
+                    .flat_map(|x| x.var_decls.clone())
+                    .collect::<Vec<_>>();
+                let arguments_buf = arguments_buf
+                    .into_iter()
+                    .map(|x| x.value)
+                    .collect::<Vec<_>>();
```

Comment:
I think you can avoid cloning here by mapping into a tuple:
```rust
let (var_decls, arguments_buf) = arguments_buf.into_iter().map(|x| (x.var_decls, x_value)).collect::<Vec<_>>();
```

Reply:
did the tuple trick not work here? I notice you resolved it and are still cloning

Reply:
(typing this on my phone). This didn't work because it results in a vec of tuples rather than a tuple of vecs. I tried a combination of map/flat_map + unzip and ended up reverting the changes and just leaving it as-is.

Reply:
You can convert a collection of pairs into a pair of collections with `unzip`.  😁 https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.unzip